lib/spack/env/cc

```diff
@@ -795,20 +839,25 @@ case "$mode" in
         if [ -n "$dtags_to_add" ] ; then
             append args_list "$linker_arg$dtags_to_add"
         fi
+        extend args_list spack_flags_rpath_dirs_list "$rpath"
```

Comment:
@trws pointed out to me that the way we're doing this here causes an extra copy of each argument beyond what could be minimized. I don't think it's worth requesting changes for that, but I wanted to note it here so we can track that.

Reply:
Greg, Tom, I completely agree with you and the only reason we're doing this is to make it easier to read and understand by processing both command line arguments and spack_flags in exactly the same way which includes duplication of `extend args_list ...` calls. We can optimize away extra copies as a separate pull request if there is such a need.